packages/graphql-mini-transforms/src/webpack.ts

```diff
@@ -3,14 +3,22 @@ import {dirname} from 'path';
 import {loader} from 'webpack';
 import {parse, DocumentNode} from 'graphql';
 
-import {cleanDocument, extractImports} from './document';
+import {cleanDocument, extractImports, CleanDocumentOptions} from './document';
 
 export default async function graphQLLoader(
   this: loader.LoaderContext,
   source: string | Buffer,
 ) {
   this.cacheable();
 
+  const options = this.query;
+
+  if (options && typeof options !== 'object') {
+    throw new Error(
+      '@shopify/graphql-mini-transforms only supports options as an object',
+    );
+  }
```

Comment:
I think the common pattern for webpack loaders it to use the loader-utils package to normalize the options.